assignment-3.md

```diff
@@ -0,0 +1,108 @@
+# Assignment 3: Tests
+
+## 3.1 Getting Started
+
+Start the test runner with `npm test`.
+
+By default only tests for changed files are run. So in case you have been making
+Git commits, and currently have no uncommitted changes in your working
+directory, press `a` to trigger a test run.
+
+The initial test in `Climate.test.tsx` probably does not run green anymore,
+because it was testing the app in its starting state, but we have changed the
+app quite a bit in the last exercises.
+
+We'll take care for that soon, but for now disable that test with Jest's `skip`
+feature. Change this:
+
+```typescript
+test('Climate app shows the current temperature…', …)
+```
+
+to this:
+
+```typescript
+test.skip('Climate app shows the current temperature…', …)
+```
+
+## 3.2 Header
+
+Let's get started by writing some tests for the `Header` component that we
+have implemented in the first assignment. You have probably put that into a
+file called `Header.tsx`. Create a new file `Header.test.tsx` for the tests.
+
+Try to `render` the component with different text values that you pass in as a
+prop. Use `getByRole` and `toHaveTextContent` to find the rendered header in the
+DOM and check its text content.
+
+*Optional:* If you're feeling fancy, you can use
+[Jest's `test.each`](https://jestjs.io/docs/en/api#testeachtablename-fn-timeout)
+to create a parametrized test.
+
+### Hints
+
+The [Testing Playground Chrome extension](https://chrome.google.com/webstore/detail/testing-playground/hejbmebodbijjdhflfknehhcgaklhano)
+can help you with writing queries.
+
+The existing test in `Climate.test.tsx` will give you an example how a typical
+test looks like. Don't copy over too much, though! The `Climate` component has
+to deal with some asynchrony, as it is listening to sensor events.
+
+Avoid using any of the following in `Header.test.tsx`:
+
+* `sensor` / `testSensor`
+* `async` / `await`
+* `findBy…` queries
+
+## 3.3 Current Temperature and Humidity
+
+Back to `Climate.test.tsx`. Remove that `skip` you added in 3.1 and get the test
+working again!
+
+When it works, add a similar test for the current humidity value.
+
+### Hints
+
+You have introduced a child component in assignment 3.2 and are using that in
```

Comment:
wasn't the child component added in assignment 2 somwhere?
3.2 is about testing the header.

Reply:
it's 2.1 I think

Reply:
Already solved in a later commit.